src/components/CardComponent/index.data.jsx

```diff
@@ -0,0 +1,20 @@
+import React, { useState, useEffect } from 'react';
+import CardComponentView from './index.view';
+
+import db from '../../db/dexie';
+
+const CardComponentData = () => {
+
+	const [data, setData] = useState("");
+	const [complete, setComplete] = useState(false);
+
+	useEffect(async () => {
+		const result = await db.card.toArray();
+		setData(result);
+		setComplete(true)
+	},[]);
+
+	return complete && <CardComponentView mainArray = {data}/>;
```

Comment:
NIT: If you are passing `data` into CardComponentView, we can assign it the same variable name (`data`) as it might make more sense.